docs/specification/draft/basic/authorization.mdx

```diff
@@ -263,35 +253,50 @@ Servers **MUST** return appropriate HTTP status codes for authorization errors:
 | 403         | Forbidden    | Invalid scopes or insufficient permissions |
 | 400         | Bad Request  | Malformed authorization request            |
 
-### 2.10 Implementation Requirements
+## 3. Security Considerations
+
+Implementations **MUST** follow OAuth 2.1 security best practices. Refer to
+[RFC9700](https://datatracker.ietf.org/doc/html/rfc9700) for details.
+
+### 3.1 Token Theft
+Attackers who obtain tokens stored by the client, by accessing tokens cached or logged on the server can access protected resources with
+requests that appear legitimate to resource servers.
+
+Clients **MUST** implement secure token storage and follow OAuth 2.0 best practices,
+as outlined in [RFC 9700](https://datatracker.ietf.org/doc/html/rfc9700).
+
+MCP authorization servers SHOULD enforce token expiration and rotation to limit the window of exploitation.
+
+### 3.2 Token Interception
+An attacker positioned between MCP clients and MCP servers can intercept tokens via [Man-in-the-Middle (MITM)](https://en.wikipedia.org/wiki/Man-in-the-middle_attack) attacks.
+
+To mitigate the risk of this threat:
+
+1. All authorization endpoints **MUST** be served over HTTPS.
+1. MCP clients **MUST** implement PKCE according to [OAuth 2.1 section 7.5.2](https://www.ietf.org/archive/id/draft-ietf-oauth-v2-1-12.html#name-countermeasures). PKCE helps  prevent authorization code interception attacks by requiring clients to create a secret verifier-challenge pair, ensuring that only the original requestor can exchange an authorization code for tokens.
+1. All redirect URIs **MUST** be either `localhost` or use HTTPS to prevent token and code interception.
+
+### 3.3 Open Redirection
```

Comment:
There is a scenario in [Section 4.11.2 of RFC9700](https://www.rfc-editor.org/rfc/rfc9700.html#section-4.11.2) that is directly related to the MCP usage of Dynamic Client Registration.

> The attacker could, for example, register a client via dynamic client registration [[RFC7591](https://www.rfc-editor.org/rfc/rfc9700.html#RFC7591)] and execute one of the following attacks:

> Intentionally send an erroneous authorization request, e.g., by using an invalid scope value, thus instructing the authorization server to redirect the user agent to its phishing site.
> ...

There are some recommendations for countermeasures in that section.

Reply:
I added a reference to this section in the paragraph below and copied over the "only automatically redirect..." bit (maybe should quote it? lmkwyt)